httpbp/client.go

```diff
@@ -0,0 +1,164 @@
+package httpbp
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/reddit/baseplate.go/tracing"
+	"github.com/sony/gobreaker"
+)
+
+// LowLevelClientMiddleware is suitable for transport-level functions.  Retries
+// and request munging should be done by wrapping the client.
+type LowLevelClientMiddleware func(name string, next http.RoundTripper) http.RoundTripper
+
+type maxConcurrency struct {
+	next           http.RoundTripper
+	activeRequests int64
+	maxConcurrency int64
+}
+
+func (m *maxConcurrency) RoundTrip(req *http.Request) (*http.Response, error) {
+	attemptedRequests := atomic.AddInt64(&m.activeRequests, 1)
+	defer atomic.AddInt64(&m.activeRequests, -1)
+	if m.maxConcurrency > 0 && attemptedRequests > m.maxConcurrency {
+		return nil, errors.New("hit concurrency limit on http round-tripper")
```

Comment:
nit: Please define this error as a global var, or at least prefix the error message with `httpbp: `.